src/www/widgets/widgets/system_log.widget.php

```diff
@@ -44,7 +44,35 @@
    exit;
 }
 
-require_once('diag_logs_common.inc');
+
+function dump_log($system_logfile, $syslogEntriesToFetch)
+{
+  global $config;
+
+  $logdatastr = configd_run("syslog dumplog {$system_logfile}");
+  $logdata = explode("\n", $logdatastr);
+  $reverse = ($config['OPNsense']['Syslog']['Reverse'] == '1');
+  if($reverse)
+    $logdata = array_reverse($logdata);
+
+  $counter = 1;
+  foreach ($logdata as $logent) {
+    if(trim($logent) == '')
+      continue;
+
+    $logent = preg_split("/\s+/", $logent, 6);
+    $entry_date_time = join(" ", array_slice($logent, 0, 3));
+    echo "<tr>\n";
```

Comment:
maybe move this line after the $entry_text assignment so the template is kept together.